### PR TITLE
Add windows and cross compile builds

### PR DESCRIPTION
**Pull request by mmcauliffe**
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mmcauliffe** [comment]
@conda-forge/reaper this should be good for review now!

FYI: I've added myself as maintainer for any issues that come up with the new architectures that have been added.

**Comment by mmcauliffe** [comment]
@conda-forge/core pining core team for review as well, since this isn't a very active feedstock

File: conda-forge.yml [hunk 1]
```diff
@@ -1,9 +1,20 @@
-compiler_stack: comp7
-max_py_ver: '37'
-max_r_ver: '35'
+azure:
+  settings_win:
+    pool:
+      vmImage: windows-2019
```
Comment (h-vetinari) [inline conda-forge.yml]: Why are you setting this to use an old windows image?

File: conda-forge.yml [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+build_platform:
+  osx_arm64: osx_64
+  linux_ppc64le: linux_64
+  linux_aarch64: linux_64
+os_version:
+  linux_64: cos7
```
Comment (h-vetinari) [inline conda-forge.yml]: That's obsolete
```suggestion
```

File: conda-forge.yml [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
 conda_forge_output_validation: true
 github:
   branch_name: main
   tooling_branch_name: main
+provider:
+  win: azure
+  linux_aarch64: azure
+  linux_ppc64le: azure
```
Comment (h-vetinari) [inline conda-forge.yml]: This is redundant with specifying the build platform (which for `linux-64` is necessarily on azure, and on win it's the default)
```suggestion
```

**Review by h-vetinari — COMMENTED** [review]
Basically LGTM aside from some obsolete/questionable configs

**Comment by mmcauliffe** [comment]
@h-vetinari thanks for the comments, cleaned up the conda forge config, my main issue is that there's not a clear section in the [knowledge base](https://conda-forge.org/docs/maintainer/updating_pkgs/) for specifically enabling other packages, there's [Configuring conda-forge.yml](https://conda-forge.org/docs/maintainer/conda_forge_yml/), but it's autogenerated or various blog posts that are no longer relevant but  come up during search, and there's a slew of options that are redundant but documented without a best practices. It might be worthwhile to have a dedicated up-to-date section for different build platforms in the contributing/maintaining packages.

**Review by h-vetinari — APPROVED** [review]
Sorry, this slipped off my radar (was travelling). You're right that the docs aren't super clear, if you have an improvement to propose, please go for it and feel free to tag me. :)

**Comment by mmcauliffe** [comment]
@conda-forge/core would it be possible to merge this in?

**Comment by h-vetinari** [comment]
Sorry, hadn't seen that you're not a maintainer yet.

**Comment by mmcauliffe** [comment]
Awesome, thanks!

**Pull request merged**